src/gurt/hash.c

```diff
@@ -1304,13 +1304,21 @@ d_hhash_link_delete(struct d_hhash *hhash, struct d_hlink *hlink)
 void
 d_hhash_link_getref(struct d_hhash *hhash, struct d_hlink *hlink)
 {
-	d_hash_rec_addref(&hhash->ch_htable, &hlink->hl_link.rl_link);
+	ch_rec_addref(&hhash->ch_htable, &hlink->hl_link.rl_link);
```

Comment:
I also think not must to add another cache layer for cont_hdl and pool hdl outside handle hash table.

But these two change in gurt/hash.c looks good as it reduced one lock taking, probably can submit the change for gurt/hash.c separately first?

Reply:
I pushed another PR for https://github.com/daos-stack/daos/pull/10249, i will push gurt/hash.c part seprately.